lib/reply.js

```diff
@@ -502,8 +502,9 @@ function handleError (reply, error, cb) {
   var res = reply.raw
   var statusCode = res.statusCode
   statusCode = (statusCode >= 400) ? statusCode : 500
+
   // treat undefined and null as same
-  if (error != null) {
+  if (error != null && (reply.context.isDefaultErrorHandler === true || reply[kReplyErrorHandlerCalled] === true)) {
```

Comment:
Can we move this behavior in the default errorHandler instead?

Reply:
If the error handler sends a new error like the default error handler does, we want to set the headers.

Reply:
It would be possible to move the behavior to default errorHandler and check reply[kReplyErrorHandlerCalled] === true where it was originally to see if we set the headers. That way Context doesn't have an additional attribute.

Reply:
I would prefer to not add another field to context, go ahead and move it.